### PR TITLE
Forge: add Scaffolding setup and generation tests

### PR DESCRIPTION
**Pull request by jrichter1**

### CODE REVIEW

**Comment by vpakan** [comment]
test this

File: tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/WizardTestBase.java [hunk 1]
```diff
@@ -213,4 +216,17 @@ public void newJPAEntity(String projectName, String entityName, String tableName
...
+	/**
+	 * Runs scaffold setup wizard on a specified project
+	 * @param projectName
+	 * @param type Type of the scaffold to be used
+	 */
+	public void scaffoldSetup(String projectName, ScaffoldType type) {
+		new ProjectExplorer().selectProjects(projectName);
+		WizardDialog dialog = getWizardDialog("Scaffold: Setup", "(Scaffold: Setup).*");
+		new DefaultCombo().setSelection(type.getName());
+		AbstractWait.sleep(TimePeriod.getCustom(2));
```
Comment (psrna) [inline tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/WizardTestBase.java]: Please don't use AbstractWait.sleep, use a wait condition instead.

File: tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/ScaffoldSetupWizardTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	/**
+	 * Tests scaffold setup for JSF
+	 */
+	@Test
+	public void testJSFScaffoldSetup() {
+		ScaffoldType type = ScaffoldType.FACES;
+		scaffoldSetup(PROJECT_NAME, type);
+		AbstractWait.sleep(TimePeriod.getCustom(2));
```
Comment (psrna) [inline tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/ScaffoldSetupWizardTest.java]: Please don't use AbstractWait.sleep, use a wait condition instead.

File: tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/ScaffoldSetupWizardTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	/**
+	 * Tests scaffold setup for AngularJS
+	 */
+	@Test
+	public void testAngularJSScaffoldSetup() {
+		ScaffoldType type = ScaffoldType.ANGULARJS;
+		scaffoldSetup(PROJECT_NAME, type);
+		AbstractWait.sleep(TimePeriod.getCustom(2));
```
Comment (psrna) [inline tests/org.jboss.tools.forge.ui.bot.test/src/org/jboss/tools/forge2/ui/bot/wizard/test/ScaffoldSetupWizardTest.java]: Please don't use AbstractWait.sleep, use a wait condition instead.

**Comment by jrichter1** [comment]
I looked over the waits and two of them were in fact redundant. The scaffoldSetup method would sometimes fail to finish correctly if the button was pressed right away, so I made it go to the next page instead and then finish.

**Comment by psrna** [comment]
I've run the tests in the suite and got an exception:
```
org.jboss.reddeer.eclipse.exception.EclipseLayerException: Cannot get project item specified by path.Project item either does not exist or solution is ambiguous because of existence of more items on the path with same name without decorators
	at org.jboss.reddeer.eclipse.core.resources.AbstractExplorerItem.getProjectItem(AbstractExplorerItem.java:173)
	at org.jboss.tools.forge2.ui.bot.wizard.test.ScaffoldWizardTest.checkWebResources(ScaffoldWizardTest.java:79)
	at org.jboss.tools.forge2.ui.bot.wizard.test.ScaffoldWizardTest.checkGeneratedFiles(ScaffoldWizardTest.java:70)
	at org.jboss.tools.forge2.ui.bot.wizard.test.ScaffoldWizardTest.testCreateFacesScaffold(ScaffoldWizardTest.java:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.jboss.reddeer.junit.internal.runner.RequirementsRunner$InvokeMethodWithException.evaluate(RequirementsRunner.java:233)
	at org.jboss.reddeer.junit.internal.runner.RunBefores.evaluate(RunBefores.java:79)
	at org.jboss.reddeer.junit.internal.runner.RunAfters.evaluate(RunAfters.java:64)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.jboss.reddeer.junit.internal.runner.RequirementsRunner.runChild(RequirementsRunner.java:129)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.jboss.reddeer.junit.internal.runner.RunBefores.evaluate(RunBefores.java:79)
	at org.jboss.reddeer.junit.internal.runner.FulfillRequirementsStatement.evaluate(FulfillRequirementsStatement.java:26)
	at org.jboss.reddeer.junit.internal.runner.CleanUpRequirementStatement.evaluate(CleanUpRequirementStatement.java:25)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.jboss.reddeer.junit.internal.runner.RequirementsRunner.run(RequirementsRunner.java:110)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.jboss.reddeer.eclipse.core.RemotePluginTestRunner.main(RemotePluginTestRunner.java:58)
	at org.jboss.reddeer.eclipse.core.UITestApplication.runTests(UITestApplication.java:115)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Testable$1.run(E4Testable.java:73)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by psrna** [comment]
Please rebase on current master and resolve conflicts.

**Pull request merged**